onnxruntime/core/providers/cuda/tensor/upsample.cc

```diff
@@ -230,7 +230,7 @@ Status Upsample<T>::BaseCompute(OpKernelContext* context,
           }
 
           const bool is_2D = X_dims.size() == 2;
-          const bool is_nchw = is_2D ? true : (scales[1] == 1.0f && scales[1] == 1.0f);
+          const bool is_nchw = is_2D ? true : scales[1] == 1.0f;
```

Comment:
Some other places also check if scales[0] equals to 1.0. Is it needed here?

Reply:
shall be `scales[0] == 1.0f && scales[1] == 1.0f` since the comment above "4-D inputs with the corresponding outermost 2 scale values being 1."

https://github.com/microsoft/onnxruntime/blob/0ed30dbf026b89f6f3b4d5328e436f125de69169/onnxruntime/core/providers/cuda/tensor/upsample.cc#L229